pkg/webhooks/monitoring.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhooks
+
+import (
+	"istio.io/pkg/monitoring"
+)
+
+var (
+	// webhookConfigNameTag holds the target webhook config name for the context.
+	webhookConfigNameTag = monitoring.MustCreateLabel("name")
+
+	// reasonTag holds the error reason for the context.
+	reasonTag = monitoring.MustCreateLabel("reason")
+)
+
+var (
+	metricWebhookPatchAttempts = monitoring.NewSum(
+		"webhook_patch_attempts",
```

Comment:
As per [prometheus naming standards]( https://prometheus.io/docs/practices/naming/#metric-names) these should end with _total